source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -40,7 +40,9 @@ void ShaderGraph::addInputSockets(const InterfaceElement& elem, GenContext& cont
             ShaderGraphInputSocket* inputSocket = nullptr;
             const string& portValue = port->getResolvedValueString();
             std::pair<const TypeDesc*, ValuePtr> enumResult;
-            if (context.getShaderGenerator().remapEnumeration(*port, portValue, enumResult))
+            const string& enumNames = port->getAttribute(ValueElement::ENUM_ATTRIBUTE);
+            const TypeDesc* type = TypeDesc::get(port->getType());
```

Comment:
`type` could be reused below to avoid calling `TypeDesc::get(port->getType())` again.